### PR TITLE
Remove blank Javadoc

### PR DESCRIPTION
**Pull request by jukzi**
This commit cleans up Javadoc that does not add information. It resolves ecj warnings:
 `Javadoc: Description expected after ...`
It helps to prevent future empty javadoc by disabling missingJavaDoc warnings. This resolves
 `Javadoc: Missing ...`

The modification is a result of regular expression search&replace:

in files `*.java`
 `^[\s]*\*[\s]*(@return|@param[\s]*[^\s]+|@throws[\s]*[^\s]+)\R([\s]*\*[\s]*@|[\s]*\*/\R)`
 ->`$2`
 `^([\s]*\*[\s]*\R)([\s]*\*/\R)`
 ->`$2`
 `^[\S\t]*/\*\*\R[\s]*\*/\R`
 ->``

in files `org.eclipse.jdt.core.prefs`
 `org\.eclipse\.jdt\.core\.compiler\.problem\.missingJavadoc(Comments|Tags)\=[^\s]*`
 ->`org\.eclipse\.jdt\.core\.compiler\.problem\.missingJavadoc$1\=ignore`

excluded resources:
 apitools/org.eclipse.pde.api.tools.tests/test-builder
 apitools/org.eclipse.pde.api.tools.tests/test-classes
 apitools/org.eclipse.pde.api.tools.tests/test-source
 apitools/org.eclipse.pde.api.tools.tests/tests-deltas

### CODE REVIEW

**Comment by HannesWell** [comment]
I support this clean-up, but due to the size of the change I think it can only be reviewed in samples and not in detail.
Thus I suggest to submit this in the very beginning of the next dev-cycle.

**Comment by HannesWell** [comment]
I support this clean-up, but due to the size of the change I think it can only be reviewed in samples and not in detail.
Thus I suggest to submit this in the very beginning of the next dev-cycle.

File: apitools/org.eclipse.pde.api.tools.tests/src/org/eclipse/pde/api/tools/anttasks/tests/ApiToolingApiuseAntTaskTests.java [hunk 1]
```diff
@@ -126,8 +126,6 @@ public void test3() throws Exception {
 
 	/**
 	 * Tests that a use scan will find illegal use problems that can be filtered
-	 *
```
Comment (HannesWell) [inline apitools/org.eclipse.pde.api.tools.tests/src/org/eclipse/pde/api/tools/anttasks/tests/ApiToolingApiuseAntTaskTests.java:129]: Does the removal of such empty lines survive a subsequent formatting with the current settings?
If not, please revert it.

Comment (jukzi) [reply apitools/org.eclipse.pde.api.tools.tests/src/org/eclipse/pde/api/tools/anttasks/tests/ApiToolingApiuseAntTaskTests.java:129]: Formatting does not add empty lines.  tested.

**Comment by jukzi** [comment]
Since the changes are only search&replace you could verify the regexp if you want. 

> Thus I suggest to submit this in the very beginning of the next dev-cycle.

The chances of merge conflict are pretty high on this PR. Therefore i'd like to use the green jenkins build to submit ASAP if there is no objection.

**Comment by HannesWell** [comment]
Jörg what's this supposed to be?
We are very late in the release cycle and the code freeze has been on Monday, which is why this check is red:
grafik

Now pushing such a huge change is not reasonable. Especially since it has zero benefit for users, but there is always a (small) risk that this breaks something especially with such large, automated changes.
In JDT you also don't push clean-ups outside of M1, even clearly before the deadline: https://github.com/eclipse-jdt/eclipse.jdt.core/pull/1403#issuecomment-1778186119

Furthermore I find it highly disrespectful to not give others participating in a discussion time to respond, just because you are in rush for whatever reason.

> > Thus I suggest to submit this in the very beginning of the next dev-cycle.
> 

> The chances of merge conflict are pretty high on this PR. Therefore i'd like to use the green jenkins build to submit ASAP if there is no objection.

No. During the quiet-period until the next release cycle only few fixes, ideally no changes happen. So the chances for conflicts are pretty low. And even if there a conflicts then, man, just resolve them or re-run the search and replace. Since you automated it, it can't be that hard.

**Comment by jukzi** [comment]
I am sorry if i disappoint you - technically you had a 24h period to raise a -1. I even waited 4h during office hours for a reply.
Regarding your JDT PR: it's totally different to change productive code that is even untested or to just modify tests or text that is tested and can't break anything - no matter what size.

**Comment by vogella** [comment]
@jukzi please do not commit without good reason in the freeze period. I personally think that the no big clean- ups in JDT is to strict. 

But official freeze periods should not be broken (not in pde nor jdt) just because we think they are not risky.

I understand and agree with you on the "high collision" risk but as the change was done via a script, it should be easy to regenerate it in my opinion if a conflict arises

**Comment by HannesWell** [comment]
> technically you had a 24h period to raise a -1. I even waited 4h during office hours for a reply.

Obviously my previous comment wasn't clear enough. Will make it more clear next time.
Furthermore you may have noticed that I usually don't spent my office-time to work on or discuss on Eclipse issues.
In general you should consider that different people respond/work on Eclipse at different times because, they live on other parts of the globe, its not their daily job, they simply have something else to do. ;)

> Regarding your JDT PR: it's totally different to change productive code that is even untested or to just modify tests or text that is tested and can't break anything - no matter what size.

Are you 100% sure you changed just text? And nothing slipped trough? Probably not, but I cannot guarantee this.
Therefore I wanted to have this submitted in early 4.31-M1.

And you might have read: https://www.eclipse.org/lists/platform-dev/msg03920.html
There are exceptions, but those exceptions have approval of two other Committers/Project-Lead, are much smaller and users of that new feature heavily benefit from it.

**Comment by HannesWell** [comment]
Created a revert PR for this now in https://github.com/eclipse-pde/eclipse.pde/pull/901.
In the beginning of the next release cycle this is a welcome change.

**Comment by HannesWell** [comment]
@jukzi do you want to re-create this PR now so that it can be submitted as soon as the master branch is open again?

**Comment by jukzi** [comment]
> @jukzi do you want to re-create this PR now so that it can be submitted as soon as the master branch is open again?

https://github.com/eclipse-pde/eclipse.pde/pull/937

**Comment by laeubi** [comment]
The master branch is not yet opened....

**Pull request merged**